### PR TITLE
test: add snap and configuration management tests

### PR DESCRIPTION
**Pull request by st3v3nmw**
This PR:

- Adds the API endpoint to get snap configuration
- Adds tests for snap & configuration management
- Adds a GitHub action to lint and test the code

Sample successful run:

image

https://github.com/st3v3nmw/snap-http/actions/runs/7424541214/job/20204390696

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    f: Callable[..., SnapdResponse] = getattr(snap_http, name)
+    response = f(*args, **kwargs)
+
+    if response.type == "sync":
+        return response
+
+    final_states = {"Done", "Abort", "Hold", "Error"}
+    change = response.change
+    while True:
+        time.sleep(1)
```
Comment (Perfect5th) [inline tests/utils.py:19]: Probably fine for now, but a minimum 1s sleep on every test might make things egregiously long eventually.

File: tests/utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+def call_and_await_api(name: str, *args: Any, **kwargs: Any) -> SnapdResponse:
+    """Call the `name` endpoint and wait until changes are applied."""
+    f: Callable[..., SnapdResponse] = getattr(snap_http, name)
+    response = f(*args, **kwargs)
+
+    if response.type == "sync":
+        return response
+
+    final_states = {"Done", "Abort", "Hold", "Error"}
```
Comment (Perfect5th) [inline tests/utils.py:16]: style nit: consider moving to global/constant set or enum. Not critical.

**Review by Perfect5th — APPROVED** [review]

File: tests/utils.py (hunk 1 above)
Comment (st3v3nmw) [reply tests/utils.py]: Reduced to 0.1s

File: tests/utils.py (hunk 2 above)
Comment (st3v3nmw) [reply tests/utils.py]: Moved to a global set

**Pull request merged**